File: server/makc2022--nodejs--nestjs/src/layer3/nosql-mongo/sample/mappers/typegoose/entities/dummy-one-to-many/mapper-dummy-one-to-many-entity.object.ts
/** Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License. */

import { modelOptions, prop } from '@typegoose/typegoose';
import { Base } from '@typegoose/typegoose/lib/defaultClasses';
import { DummyOneToManyEntityObject } from '../../../../entities/dummy-one-to-many/dummy-one-to-many-entity.object';

export interface MapperDummyOneToManyEntityObject extends Base<string> {}

/** Объект сущности "DummyManyToMany" сопоставителя. */
@modelOptions({ schemaOptions: { collection: 'dummy-one-to-many' } })
export class MapperDummyOneToManyEntityObject implements DummyOneToManyEntityObject {
  /** @inheritdoc */
  @prop({ unique: true, required: true })
  name!: string;
}

File: server/makc2022--dotnet/Makc2022.Layer3.Sql.Sample.Mappers.EF/Entities/DummyOneToMany/MapperDummyOneToManyEntityObject.cs
// Copyright (c) 2022 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

using Makc2022.Layer3.Sql.Sample.Entities.DummyOneToMany;
using Makc2022.Layer3.Sql.Sample.Mappers.EF.Entities.DummyMain;

namespace Makc2022.Layer3.Sql.Sample.Mappers.EF.Entities.DummyOneToMany
{
    /// <summary>
    /// Объект сущности "DummyOneToMany" сопоставителя.
    /// </summary>
    public class MapperDummyOneToManyEntityObject : DummyOneToManyEntityObject
    {
        #region Properties

        /// <summary>
        /// Объекты сущности "DummyMain".
        /// </summary>
        public List<MapperDummyMainEntityObject> ObjectsOfDummyMainEntity { get; } =
            new List<MapperDummyMainEntityObject>();

        #endregion Properties
    }
}
